styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -150,13 +153,14 @@ public String createExecution(final RunState runState, final String execName, fi
         .filter(mode -> mode != ExecutionMode.UNRECOGNIZED)
         .orElseThrow(() -> new CreateExecutionException("Missing trigger or unknown in StateData: " + runState.data()));
 
-    var styxVariables = getStyxVariables(runState.workflowInstance(), runState.data());
-    var triggeredParams = runState.data().triggerParameters()
+    final var styxVariables = getStyxVariables(runState.workflowInstance(), runState.data());
+    final var triggeredParams = runState.data().triggerParameters()
         .map(FlyteAdminClientRunner::getFilteredTriggerParams)
         .orElseGet(Map::of);
 
-    // labels values should be normalized calling LabelValue::normalize, but styx execution ids are normalized already
-    var labels = Map.of(STYX_EXECUTION_ID, styxVariables.get(STYX_EXECUTION_ID));
+    final var labels = styxVariables.entrySet().stream()
+        .map(entry -> Map.entry(entry.getKey(), LabelValue.normalize(entry.getValue())))
+        .collect(collectingAndThen(toMap(Map.Entry::getKey, Map.Entry::getValue), ImmutableMap::copyOf));
```

Comment:
:art: you could avoid the copying here and use `toUnmodifiableMap`

```suggestion
        .collect(toUnmodifiableMap(Map.Entry::getKey, Map.Entry::getValue));
```